front/components/assistant/conversation/ConversationContainer.tsx

```diff
@@ -0,0 +1,354 @@
+import { Page } from "@dust-tt/sparkle";
+import type {
+  AgentMention,
+  AgentMessageWithRankType,
+  LightAgentConfigurationType,
+  MentionType,
+  SubscriptionType,
+  UserMessageWithRankType,
+  UserType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { Transition } from "@headlessui/react";
+import { cloneDeep } from "lodash";
+import { useRouter } from "next/router";
+import {
+  Fragment,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+import { ReachedLimitPopup } from "@app/components/app/ReachedLimitPopup";
+import { AssistantBrowserContainer } from "@app/components/assistant/conversation/AssistantBrowerContainer";
+import ConversationViewer from "@app/components/assistant/conversation/ConversationViewer";
+import { FixedAssistantInputBar } from "@app/components/assistant/conversation/input_bar/InputBar";
+import { InputBarContext } from "@app/components/assistant/conversation/input_bar/InputBarContext";
+import type { ContentFragmentInput } from "@app/components/assistant/conversation/lib";
+import {
+  createConversationWithMessage,
+  createPlaceholderUserMessage,
+  submitMessage,
+} from "@app/components/assistant/conversation/lib";
+import { NewConversationWrapper } from "@app/components/assistant/conversation/NewConversationWrapper";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import type { FetchConversationMessagesResponse } from "@app/lib/api/assistant/messages";
+import { getRandomGreetingForName } from "@app/lib/client/greetings";
+import { useSubmitFunction } from "@app/lib/client/utils";
+import { useConversationMessages } from "@app/lib/swr";
+
+interface ConversationContainerProps {
+  conversationId: string | null;
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  user: UserType;
+}
+
+export function ConversationContainer({
```

Comment:
It's not just about creating a component for reuse. There are many more benefits! For instance, it helps manage which parts of the app are re-rendered when a new conversation is initiated. We assign a key to this component, which limits re-rendering to when the key changes. The other option would be to assign keys to all components and engage in extensive prop drilling, which, in my opinion, is less effective than using a clear, distinct component.

Reply:
One should not see relevance to create a component because it will be re-used. They are much more benefits to component. In this case, it's needed so we can control which part of the app is re-rendered when we create a new conversation. If you look here, we pass a key to this component which constraint re-rendering when the key changes.

The alternative to this is setting keys to all component and do some props drilling which IMHO is worst than clear component.